api/app/controllers/spree/api/v2/storefront/cart_controller.rb

```diff
@@ -161,6 +161,9 @@ def associate
           def change_currency
             spree_authorize! :update, spree_current_order, order_token
 
+            spree_current_order.ensure_updated_shipments
```

Comment:
This logic should be moved to `change_currency_service`, to make the behavior consistent for all users of change currency service